test/ModAttachr/modAttachr.ts

```diff
@@ -0,0 +1,71 @@
+import { EventNames } from "../../src/EventNames";
+import { IMod } from "../../src/IModAttachr";
+import { ModAttachr } from "../../src/ModAttachr";
+import { mochaLoader } from "../main";
+import { CustomEventNames } from "../utils/fakes";
+
+mochaLoader.it("Testing to ensure onModEnable is fired properly", (): void => {
+    // Arrange
+    const eventNames: EventNames = new EventNames();
+    const dummyMod: IMod = {
+        name: "Dummy Mod",
+        events: {
+            onModEnable: (): string => {
+                return "success";
+            }
+        },
+        enabled: false
+    };
+    const modAttachr: ModAttachr = new ModAttachr({
+        mods: [dummyMod],
+        storeLocally: false,
+        eventNames: eventNames
+    });
+
+    // Assert
+    chai.expect(modAttachr.fireModEvent(modAttachr.eventNames.onModEnable, "Dummy Mod")).to.be.equal("success");
```

Comment:
The AAA pattern is great because it distinctly separates the three phases of a good unit test. Here you're mixing acting and asserting. Store the result of `fireModEvent` in a variable as part of the act step, then assert it's what's expected as the assert step.